From 99f2c35d570573bfdf5ef56fcebec93f75cc1054 Mon Sep 17 00:00:00 2001
From: wochap <gean.marroquin@gmail.com>
Date: Tue, 12 Mar 2024 16:40:25 -0500
Subject: [PATCH 1/3] feat: print visible_appids

app_id of clients visible on the current monitor/tags
---
 dwl.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/dwl.c b/dwl.c
index ef27a1d..d408efd 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1,3 +1,4 @@
+#define _GNU_SOURCE
 /*
  * See LICENSE file for copyright and license details.
  */
@@ -1821,12 +1822,16 @@ printstatus(void)
 	Client *c;
 	uint32_t occ, urg, sel;
 	const char *appid, *title;
+	char *visible_appids = NULL;
 
 	wl_list_for_each(m, &mons, link) {
 		occ = urg = 0;
 		wl_list_for_each(c, &clients, link) {
 			if (c->mon != m)
 				continue;
+			if (VISIBLEON(c, m)) {
+				asprintf(&visible_appids, "%s%s ", visible_appids ? visible_appids : "", client_get_appid(c));
+			}
 			occ |= c->tags;
 			if (c->isurgent)
 				urg |= c->tags;
@@ -1851,6 +1856,9 @@ printstatus(void)
 		printf("%s tags %u %u %u %u\n", m->wlr_output->name, occ, m->tagset[m->seltags],
 				sel, urg);
 		printf("%s layout %s\n", m->wlr_output->name, m->ltsymbol);
+
+		printf("%s visible_appids %s\n", m->wlr_output->name, visible_appids ? visible_appids : "");
+		free(visible_appids);
 	}
 	fflush(stdout);
 }
-- 
2.43.2


From 19309442285cea5f43ce9e819f8ed8c6b0fa55fa Mon Sep 17 00:00:00 2001
From: wochap <gean.marroquin@gmail.com>
Date: Tue, 12 Mar 2024 18:47:14 -0500
Subject: [PATCH 2/3] feat: add focused value next to appid

---
 dwl.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/dwl.c b/dwl.c
index d408efd..f41b49b 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1820,17 +1820,19 @@ printstatus(void)
 {
 	Monitor *m = NULL;
 	Client *c;
+	Client *csel;
 	uint32_t occ, urg, sel;
 	const char *appid, *title;
 	char *visible_appids = NULL;
 
 	wl_list_for_each(m, &mons, link) {
 		occ = urg = 0;
+	  csel = focustop(m);
 		wl_list_for_each(c, &clients, link) {
 			if (c->mon != m)
 				continue;
 			if (VISIBLEON(c, m)) {
-				asprintf(&visible_appids, "%s%s ", visible_appids ? visible_appids : "", client_get_appid(c));
+				asprintf(&visible_appids, "%s%s %s ", visible_appids ? visible_appids : "", client_get_appid(c), c == csel ? "true" : "false");
 			}
 			occ |= c->tags;
 			if (c->isurgent)
-- 
2.43.2


From 408fb043b5d3b6fca4cecb6d9c55deac2101c1d7 Mon Sep 17 00:00:00 2001
From: wochap <gean.marroquin@gmail.com>
Date: Mon, 8 Apr 2024 15:40:34 -0500
Subject: [PATCH 3/3] fix `free` crashing dwl

---
 dwl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dwl.c b/dwl.c
index f41b49b..67e5e9b 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1823,11 +1823,12 @@ printstatus(void)
 	Client *csel;
 	uint32_t occ, urg, sel;
 	const char *appid, *title;
-	char *visible_appids = NULL;
+	char *visible_appids;
 
 	wl_list_for_each(m, &mons, link) {
 		occ = urg = 0;
 	  csel = focustop(m);
+		visible_appids = NULL;
 		wl_list_for_each(c, &clients, link) {
 			if (c->mon != m)
 				continue;
-- 
2.43.2

